buildSrc/src/main/groovy/org/apache/beam/gradle/GrpcVendoring_1_36_0.groovy

```diff
@@ -70,7 +70,6 @@ class GrpcVendoring_1_36_0 {
     return [
       'com.google.errorprone:error_prone_annotations:2.4.0',
       'commons-logging:commons-logging:1.2',
-      'org.apache.logging.log4j:log4j-api:2.6.2',
```

Comment:
I don't think this is what we wanted to do.

The idea has always been to have the vendored libraries expose some runtime deps as not everything should be relocated (e.g. logging shouldn't be relocated otherwise we lose logging from the relocated code). We should have just bumped the version to something that doesn't have the security issue and is compatible with the 2.6.2 version.

Ditto on the exclusions, we specifically keep them to prevent relocating logging stuff and other libs that can't be relocated.

Reply:
I created a pom.xml containing the grpc 1.36.0 and its transitive deps and it doesn't look like log4j appears in them anymore which is great and that means this change is good as is.

pom.xml
```
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
  <modelVersion>4.0.0</modelVersion>
  <groupId>com.mycompany.app</groupId>
  <artifactId>my-app</artifactId>
  <version>1.0-SNAPSHOT</version>
  <properties>
    <maven.compiler.source>1.8</maven.compiler.source>
    <maven.compiler.target>1.8</maven.compiler.target>
  </properties>
  <dependencies>
    <dependency>
      <groupId>com.google.guava</groupId>
      <artifactId>guava</artifactId>
      <version>30.1-jre</version>
    </dependency>
    <dependency>
      <groupId>com.google.protobuf</groupId>
      <artifactId>protobuf-java</artifactId>
      <version>3.15.3</version>
    </dependency>
    <dependency>
      <groupId>com.google.protobuf</groupId>
      <artifactId>protobuf-java-util</artifactId>
      <version>3.15.3</version>
    </dependency>
    <dependency>
      <groupId>com.google.code.gson</groupId>
      <artifactId>gson</artifactId>
      <version>2.8.6</version>
    </dependency>
    <dependency>
      <groupId>io.grpc</groupId>
      <artifactId>grpc-auth</artifactId>
      <version>1.36.0</version>
    </dependency>
    <dependency>
      <groupId>io.grpc</groupId>
      <artifactId>grpc-core</artifactId>
      <version>1.36.0</version>
    </dependency>
    <dependency>
      <groupId>io.grpc</groupId>
      <artifactId>grpc-context</artifactId>
      <version>1.36.0</version>
    </dependency>
    <dependency>
      <groupId>io.grpc</groupId>
      <artifactId>grpc-netty</artifactId>
      <version>1.36.0</version>
      <scope>test</scope>
    </dependency>
    <dependency>
      <groupId>io.grpc</groupId>
      <artifactId>grpc-protobuf</artifactId>
      <version>1.36.0</version>
    </dependency>
    <dependency>
      <groupId>io.grpc</groupId>
      <artifactId>grpc-stub</artifactId>
      <version>1.36.0</version>
    </dependency>
    <dependency>
      <groupId>io.netty</groupId>
      <artifactId>netty-transport-native-epoll</artifactId>
      <version>4.1.52.Final</version>
      <classifier>linux-x86_64</classifier>
    </dependency>
    <dependency>
      <groupId>io.netty</groupId>
      <artifactId>netty-tcnative-boringssl-static</artifactId>
      <version>2.0.34.Final</version>
    </dependency>
    <dependency>
      <groupId>com.google.auth</groupId>
      <artifactId>google-auth-library-credentials</artifactId>
      <version>0.22.2</version>
    </dependency>
    <dependency>
      <groupId>io.grpc</groupId>
      <artifactId>grpc-testing</artifactId>
      <version>1.36.0</version>
    </dependency>
    <dependency>
      <groupId>com.google.api.grpc</groupId>
      <artifactId>proto-google-common-protos</artifactId>
      <version>2.0.1</version>
    </dependency>
    <dependency>
      <groupId>io.opencensus</groupId>
      <artifactId>opencensus-api</artifactId>
      <version>0.28.0</version>
    </dependency>
    <dependency>
      <groupId>io.opencensus</groupId>
      <artifactId>opencensus-contrib-grpc-metrics</artifactId>
      <version>0.28.0</version>
    </dependency>
    <dependency>
      <groupId>org.conscrypt</groupId>
      <artifactId>conscrypt-openjdk-uber</artifactId>
      <version>2.5.1</version>
    </dependency>
  </dependencies>
</project>
```

mvn dependency:tree
```
[INFO] com.mycompany.app:my-app:jar:1.0-SNAPSHOT
[INFO] +- com.google.guava:guava:jar:30.1-jre:compile
[INFO] |  +- com.google.guava:failureaccess:jar:1.0.1:compile
[INFO] |  +- com.google.guava:listenablefuture:jar:9999.0-empty-to-avoid-conflict-with-guava:compile
[INFO] |  +- com.google.code.findbugs:jsr305:jar:3.0.2:compile
[INFO] |  +- org.checkerframework:checker-qual:jar:3.5.0:compile
[INFO] |  +- com.google.errorprone:error_prone_annotations:jar:2.3.4:compile
[INFO] |  \- com.google.j2objc:j2objc-annotations:jar:1.3:compile
[INFO] +- com.google.protobuf:protobuf-java:jar:3.15.3:compile
[INFO] +- com.google.protobuf:protobuf-java-util:jar:3.15.3:compile
[INFO] +- com.google.code.gson:gson:jar:2.8.6:compile
[INFO] +- io.grpc:grpc-auth:jar:1.36.0:compile
[INFO] |  +- io.grpc:grpc-api:jar:1.36.0:compile (version selected from constraint [1.36.0,1.36.0])
[INFO] |  \- org.codehaus.mojo:animal-sniffer-annotations:jar:1.19:runtime
[INFO] +- io.grpc:grpc-core:jar:1.36.0:compile
[INFO] |  +- com.google.android:annotations:jar:4.1.1.4:runtime
[INFO] |  \- io.perfmark:perfmark-api:jar:0.23.0:runtime
[INFO] +- io.grpc:grpc-context:jar:1.36.0:compile
[INFO] +- io.grpc:grpc-netty:jar:1.36.0:test
[INFO] |  +- io.netty:netty-codec-http2:jar:4.1.52.Final:test
[INFO] |  |  +- io.netty:netty-codec:jar:4.1.52.Final:test
[INFO] |  |  +- io.netty:netty-handler:jar:4.1.52.Final:test
[INFO] |  |  \- io.netty:netty-codec-http:jar:4.1.52.Final:test
[INFO] |  \- io.netty:netty-handler-proxy:jar:4.1.52.Final:test
[INFO] |     \- io.netty:netty-codec-socks:jar:4.1.52.Final:test
[INFO] +- io.grpc:grpc-protobuf:jar:1.36.0:compile
[INFO] |  \- io.grpc:grpc-protobuf-lite:jar:1.36.0:compile
[INFO] +- io.grpc:grpc-stub:jar:1.36.0:compile
[INFO] +- io.netty:netty-transport-native-epoll:jar:linux-x86_64:4.1.52.Final:compile
[INFO] |  +- io.netty:netty-common:jar:4.1.52.Final:compile
[INFO] |  +- io.netty:netty-buffer:jar:4.1.52.Final:compile
[INFO] |  +- io.netty:netty-transport:jar:4.1.52.Final:compile
[INFO] |  |  \- io.netty:netty-resolver:jar:4.1.52.Final:compile
[INFO] |  \- io.netty:netty-transport-native-unix-common:jar:4.1.52.Final:compile
[INFO] +- io.netty:netty-tcnative-boringssl-static:jar:2.0.34.Final:compile
[INFO] +- com.google.auth:google-auth-library-credentials:jar:0.22.2:compile
[INFO] +- io.grpc:grpc-testing:jar:1.36.0:compile
[INFO] |  \- junit:junit:jar:4.12:compile
[INFO] |     \- org.hamcrest:hamcrest-core:jar:1.3:compile
[INFO] +- com.google.api.grpc:proto-google-common-protos:jar:2.0.1:compile
[INFO] +- io.opencensus:opencensus-api:jar:0.28.0:compile
[INFO] +- io.opencensus:opencensus-contrib-grpc-metrics:jar:0.28.0:compile
[INFO] \- org.conscrypt:conscrypt-openjdk-uber:jar:2.5.1:compile
```

Reply:
Created https://github.com/apache/beam/pull/15113 to address this.

Reply:
Turns out netty attempts to load various loggers so this change was always ok. #15113 removes the additional logging libs.